flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/ExecutionGraph.java

```diff
@@ -732,6 +758,18 @@ else if (current == JobStatus.RESTARTING) {
 		}
 	}
 
+	public void stop() {
+		if(jobType == JobType.STREAMING) {
+			for(ExecutionVertex ev : this.getAllExecutionVertices()) {
+				if(ev.getNumberOfInputs() == 0) { // send signal to sources only
+					ev.stop();
+				}
+			}
+		} else {
+			throw new RuntimeException("STOP is only supported by streaming jobs.");
```

Comment:
But I think that it is a good idea to introduce a `StoppingException` because it's definitely not a `RuntimeException`.